xml/System.CodeDom.Compiler/TempFileCollection.xml

```diff
@@ -552,8 +552,8 @@
         <remarks>
           <format type="text/markdown"><![CDATA[  
   
-## Remarks  
- This method is called by the public `Dispose()` method and the <xref:System.Object.Finalize%2A> method. `Dispose()` invokes the protected `Dispose(Boolean)` method with the `disposing` parameter set to `true`. <xref:System.Object.Finalize%2A> invokes `Dispose` with `disposing` set to `false`.  
+## Remarks
+ This method is called by the public `Dispose()` method and the <xref:System.Object.Finalize> method, if it has been overridden. `Dispose()` invokes this method with the `disposing` parameter set to `true`. `Finalize` invokes this method with `disposing` set to `false`.
```

Comment:
This remark isn't generic boilerplate but is actually the text on the TempFileCollection page.  We know whether TempFileCollection implements a finalizer (it does).  Seems like it would make sense to specialize this with that knowledge?  Same for the rest of the occurrences here where we know whether it does or doesn't implement a finalizer.

Reply:
Do you mean specializing this general text in both cases? (when finalizer is overridden VS when it's missing)

What if we change the implementation (delete or add a finalizer)? Are we sure we have the processes, the awareness and the capacity to always notice such a change and change the documentation accordingly? Also, I would find a specialized text confusing, in case the finalizer is *missing*. `Dispose(false)` is never invoked from the built-in hierarchy, that case only makes sense if a subclass overrides the finalizer. We would need to properly explain this in the remarks section.

IMO the boilerplate text is a safer option if we are not sure if we can address these concerns.